optional.hpp

```diff
@@ -617,7 +617,7 @@ class optional<T&>
   
   constexpr optional(nullopt_t) noexcept : ref(nullptr) {}
    
-  constexpr optional(T& v) noexcept : ref(detail_::static_addressof(v)) {}
+  constexpr optional(T& v) noexcept : ref(&v) {}
```

Comment:
Let's implement this with an ifdef, e.g.

```
#ifdef OPTIONAL_CONDITIONAL_REF
... old code ...
#else
... new code ...
#endif

Reply:
If the problem is the `detail_::static_addressof` overloads lets get rid of them entirely and replace it with `std::addressof`. Using `std::addressof` is important in case the `&` operator is overloaded by `T`